### PR TITLE
Django1.11 updates

### PR DESCRIPTION
**Pull request by irabinovitch**
update scalereg to work with django 1.11

### CODE REVIEW

File: scalereg/simple_cfp/views.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from django.conf import settings
-from django.forms.util import ErrorList
+from django.forms.utils import ErrorList
```
Comment (leizleiz) [inline scalereg/simple_cfp/views.py]: WDYT about just dropping the apps we are not using to reduce the maintenance burden?

File: scalereg/reg6/staff/views.py [hunk 2]
```diff
@@ -181,6 +201,90 @@ def CashPayment(request):
     })
 
 
+@login_required
+def CashPaymentRegistered(request):
```
Comment (leizleiz) [inline scalereg/reg6/staff/views.py]: This commit contains a lot of unrelated changes. We should separate them to make the code clearer. Should I upstream some of this code first?

File: scalereg/settings.py [hunk 3]
```diff
@@ -84,18 +77,21 @@
...
-
-TEMPLATE_CONTEXT_PROCESSORS = (
-    'django.contrib.messages.context_processors.messages',
-    'django.contrib.auth.context_processors.auth',
-)
+TEMPLATES = [
+  {
+    'BACKEND': 'django.template.backends.django.DjangoTemplates',
+    'DIRS': [
+      '/var/www/django/scalereg/shellconf_templates',
```
Comment (leizleiz) [inline scalereg/settings.py]: Let's keep the entries here completely bogus like before, so it is obvious that one need to change it.

**Comment by leizleiz** [comment]
Looking at the Django roadmap, this is only going to hold us over through mid-2020. We'll need to move to Django 2.x after that. I will likely start a new Python 3 version of scalereg for that, since Django has changed quite a bit. Django 2.x does not even use the same directory structure for apps anymore.

File: scalereg/reg6/staff/views.py (hunk 2 above)
Comment (irabinovitch) [reply scalereg/reg6/staff/views.py]: If you can that'd be great.

File: scalereg/settings.py (hunk 3 above)
Comment (irabinovitch) [reply scalereg/settings.py]: Missed this one. Will clean it up.

**Comment by irabinovitch** [comment]
Definitely not a long term solution, but gets us through SCALE, ShellCon and a few other events.  We should definitely rewrite on django2 / python3.

File: scalereg/simple_cfp/views.py (hunk 1 above)
Comment (irabinovitch) [reply scalereg/simple_cfp/views.py]: im not using these elsewhere either.  happy to drop them.

Comment (irabinovitch) [reply scalereg/simple_cfp/views.py]: rebased on your PR, so these are gone now.

File: scalereg/reg6/staff/views.py (hunk 2 above)
Comment (leizleiz) [reply scalereg/reg6/staff/views.py]: Working on that. There are a few dependencies, so some PRs will have to be merged before others. Hopefully in a day or two.

Comment (irabinovitch) [reply scalereg/reg6/staff/views.py]: rebased on master, which should have this addressed now based on your other PRs.

File: scalereg/reg6/staff/views.py [hunk 4]
```diff
@@ -33,9 +33,9 @@ def CheckIn(request):
       })
 
   attendees = []
-  if request.POST['verisign']:
+  if request.POST.get('verisign'):
```
Comment (leizleiz) [inline scalereg/reg6/staff/views.py]: BTW, I think request.POST['key_name'] should still work in 1.11 just like before. If there is a MultiValueDictKeyError, that is likely a separate issue. We can deal with that in a separate PR and add the appropriate 'key_name' in request.POST check.

File: scalereg/reports/urls.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 from django.conf.urls import *
 from scalereg.reg6 import models
+from scalereg.reports import views as reports_views
```
Comment (leizleiz) [inline scalereg/reports/urls.py]: Needs to be used below?

File: scalereg/reg6/staff/views.py (hunk 4 above)
Comment (irabinovitch) [reply scalereg/reg6/staff/views.py]: The distinction is just that request.POST[] will return a keyerror or multidict error if the variable in'st present.  request.POST.get returns NONE or a default value if you provide one.   

In one or two places (i can't recall where specifically) we were seeing failures because of headers that were optional.  Can pull this out and make a 2nd PR, but this seems like the behavior we'd want.

Comment (leizleiz) [reply scalereg/reg6/staff/views.py]: Let's leave it alone. Exceptions being throwing is a good thing. It means some places are not checking the keys like they should. Converting them to get() is just masking the failure. As an example of a check, look at CheckPaymentAmount() in reg6/views.py. Before accessing request.POST['AMOUNT'], the code first calls CheckVars() to make sure the 'AMOUNT' key exists. There is no need to convert that to get().

Also, note that the staff pages may be less robust since they are not publicly accessible.
